### PR TITLE
4731 squeeze dim update meta

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #4731

(additionally fixes https://github.com/Project-MONAI/tutorials/issues/569)

### Description
squeeze dim drops a spatial axis should also update the affine

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utility/array.py [hunk 1]
```diff
@@ -624,11 +626,23 @@ def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
...
         # for pytorch/numpy unification
-        if img.shape[self.dim] != 1:
-            raise ValueError(f"Can only squeeze singleton dimension, got shape {img.shape}.")
-        return img.squeeze(self.dim)
+        if img.shape[dim] != 1:
+            raise ValueError(f"Can only squeeze singleton dimension, got shape {img.shape[dim]} of {img.shape}.")
+        img = img.squeeze(dim)
+        if self.update_meta and isinstance(img, MetaTensor) and dim > 0 and len(img.affine.shape) == 2:
+            h, w = img.affine.shape
+            affine, device = img.affine, img.affine.device if isinstance(img.affine, torch.Tensor) else None
```
Comment (KumoLiu) [inline monai/transforms/utility/array.py:639]: Do we need some warning if the input image is not in axial orientation？

Comment (wyli) [reply monai/transforms/utility/array.py:639]: No, this is about squeezing any one of the spatial dimensions, I'll make an example in the discussion.

Comment (KumoLiu) [reply monai/transforms/utility/array.py:639]: Hi @wyli, 
Maybe I didn't express myself very clearly, I know that `SqueezeDim` can squeeze in any one of the spatial dimensions, but if the user doesn't turn the image into the axial orientation, then the updated affine will also be incorrect. I'm not sure if the warning needs to be in this function, or maybe not.
```
original affine
tensor([[ -0.7422,   0.0000,   0.0000, 186.3210],
        [  0.0000,   0.0000,  -3.0000,  70.6580],
        [  0.0000,  -0.7422,   0.0000, 189.4130],
        [  0.0000,   0.0000,   0.0000,   1.0000]], dtype=torch.float64)
with orientation
tensor([[   0.7422,    0.0000, -192.9432],
        [   0.0000,    0.7422, -189.8512],
        [   0.0000,    0.0000,    1.0000]])
without orientation
tensor([[ -0.7422,   0.0000, 186.3210],
        [  0.0000,   0.0000, 189.4130],
        [  0.0000,   0.0000,   1.0000]], dtype=torch.float64)
```

Comment (wyli) [reply monai/transforms/utility/array.py:639]: I'm not sure how to warn properly within this transform, we couldn't assume orientation here. But I'll add a determinant value check, if it's zero it's probably problematic.

**Review by KumoLiu — APPROVED** [review]
Look good to me, thanks!

**Pull request merged**